src/gui/src/heatMap.cpp

```diff
@@ -583,43 +622,92 @@ void HeatMapRenderer::drawObjects(Painter& painter)
 
   const odb::Rect& bounds = painter.getBounds();
 
-  for (const auto& [bbox, map_pt] : datasource_.getMap()) {
-    if (!map_pt->has_value) {  // value not set so nothing to draw
+  // minimum box size is 2 pixels
+  const double min_dbu = 2.0 / painter.getPixelsPerDBU();
+
+  const double dbu_per_micron = datasource_.getBlock()->getDbUnitsPerMicron();
+  const int x_scale
+      = std::ceil(min_dbu / (datasource_.getGridXSize() * dbu_per_micron));
+  const int y_scale
+      = std::ceil(min_dbu / (datasource_.getGridYSize() * dbu_per_micron));
+
+  std::vector<HeatMapDataSource::MapColor> draw_pts;
```

Comment:
reserve at x_size * y_size ?

Can this be drawn immediately after down sampling rather than storing and later drawing?

Reply:
yes, that would also be a good idea. I didn't think of it when doing this.

Reply:
I'm ok to go ahead with this as it already has benefit.  We can revisit it later if needed.  I think the reserve should be easy enough to add this to PR.

Reply:
Fixing the draw loop should be a pretty minor move of the code, I just didn't consider it. I'll have something in soon-ish

Reply:
@maliberty I made the change